resources/clj-kondo.exports/tensegritics/clojuredart/hooks/flutter.clj

```diff
@@ -0,0 +1,42 @@
+(ns hooks.flutter
+  (:require [clj-kondo.hooks-api :as api]
+            [clojure.set :as set]
+            [clojure.string :as str]))
+
+(defn widget 
+  "Hook for cljd.flutter.alpha/widget macro"
+  [{:keys [:node]}]
+    (let [args (rest (:children node))
+          opts-nodes (take-while (comp api/keyword-node? first) (partition 2 args))
+          [_ state] (first (filter #(-> % first (api/sexpr) (= :state)) opts-nodes))
+
+          bindings-count (* 2 (count opts-nodes))
+          body (drop bindings-count args)
+          error (fn [msg] (throw (ex-info msg {})))
+          unknown-keys (set/difference 
+                         (into #{} (map (comp api/sexpr first) opts-nodes))
+                         #{:state :key :watch :context})]
+
+      (cond
+        (and state (not (api/vector-node? state)))
+        (error "State should be a vector [name initial-value]")
+
+        (and state (not= (count (:children state)) 2))
+        (error "State should contain only one name-value pair")
+
+        (seq unknown-keys)
+        (error (str "Unknown keys " (str/join " " unknown-keys)))
+
+        (not= 1 (- (count args) bindings-count)) ;; body is the one that remains 
```

Comment:
body length can be anything, even if it will be most of the time 1, it may be more in case of side-effects, a length of 0 would be most certainly an error.

Reply:
I guess, something like this will do the job:
```clojure
:else 
        {:node (api/list-node 
                 (list
                   (api/token-node 'let)
                   (api/vector-node (:children state))
                   watch
                   key
                   body))}
```
Only context allow to generate new symbols
<img width="361" alt="image" src="https://user-images.githubusercontent.com/14236531/164208810-22c8f4a6-7b53-4481-be25-abadd538b19f.png">

Reply:
<img width="482" alt="image" src="https://user-images.githubusercontent.com/14236531/164209241-393d5bce-451f-46b4-9176-dedb53bc02af.png">

Reply:
I will see what else could be done, today evening